File: function.scala
// function.scala
object FunctionExample {
  def myFunction(param: Int): Int = {
    var x = param
    x += 10
    if (x > 15) {
      x -= 5
    } else {
      x += 5
    }
    for (i <- 0 until 3) {
      x += i
    }
    x
  }

  def main(args: Array[String]): Unit = {
    print("Enter an integer: ")
    val input = scala.io.StdIn.readLine()
    val param = input.toInt
    if (param == 0 && input != "0") {
      println("Invalid input")
    } else {
      println(myFunction(param))
    }
  }
}

File: function.nim
# function.nim
proc myFunction(param: int): int =
    var x = param
    x += 10
    if x > 15:
        x -= 5
    else:
        x += 5
    for i in 0..2:
        x += i
    return x

echo "Enter an integer: "
let input = readLine(stdin)
let param = parseInt(input)
if param == 0 and input != "0":
    echo "Invalid input"
else:
    echo myFunction(param)
